topology/connection_creator.cpp

```diff
@@ -131,8 +137,17 @@ ConnectionCreator::ConnectionCreator( DictionaryDatum dict )
   }
   if ( not delay_.valid() )
   {
-    delay_ =
-      TopologyModule::create_parameter( ( *syn_defaults )[ names::delay ] );
+    if ( synapse_model_
+      == static_cast< index >( kernel().model_manager.get_synapsedict()->lookup(
+           "gap_junction" ) ) )
```

Comment:
Look-up by name is unsafe when users copy models or new gap-type models are added. `syn_defaults[names::has_delay]` is false for connections that have no delay.